### PR TITLE
feat: Add net6 mobile (WinUI) target for sample

### PR DESCRIPTION
**Pull request by nickrandolph**
## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

No sample for net6 targets (ios, android, macos, catalyst)

## What is the new behavior?

Added a Mobile project under the WinUI samples that has the missing targets

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: samples/Uno.Toolkit.WinUI.Samples/Uno.Toolkit.WinUI.Samples.Mobile/Android/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.companyname.Uno.Toolkit.WinUI.Samples" android:versionCode="1" android:versionName="1.0">
```
Comment (kazo0) [inline samples/Uno.Toolkit.WinUI.Samples/Uno.Toolkit.WinUI.Samples.Mobile/Android/AndroidManifest.xml:2]: I'm wondering if the package name should be the same as the WinUI Xamarin.Android one (Uno.Toolkit.WinUI.Samples). Similar to how it looks to be done for Gallery

Xamarin:
https://github.com/unoplatform/Uno.Gallery/blob/afb97a5f511ef703d5676382e841f9295a3ba9ba/Uno.Gallery/Uno.Gallery.Droid/Properties/AndroidManifest.xml#L2

net6.0-android:
https://github.com/unoplatform/Uno.Gallery/blob/afb97a5f511ef703d5676382e841f9295a3ba9ba/Uno.Gallery/Uno.Gallery.Mobile/Android/AndroidManifest.xml#L2

Not sure if this really matters though

File: samples/Uno.Toolkit.WinUI.Samples/Uno.Toolkit.WinUI.Samples.Mobile/iOS/Info.plist [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+	<key>CFBundleDisplayName</key>
+	<string>Uno.Toolkit.WinUI.Samples</string>
+	<key>CFBundleIdentifier</key>
+	<string>com.companyname.Uno.Toolkit.WinUI.Samples</string>
```
Comment (kazo0) [inline samples/Uno.Toolkit.WinUI.Samples/Uno.Toolkit.WinUI.Samples.Mobile/iOS/Info.plist:8]: I'm wondering if the Bundle ID should be the same as the WinUI Xamarin.iOS one (uno.platform.toolkit). Similar to how it looks to be done for Gallery

Xamarin:
https://github.com/unoplatform/Uno.Gallery/blob/afb97a5f511ef703d5676382e841f9295a3ba9ba/Uno.Gallery/Uno.Gallery.iOS/Info.plist#L8

net6.0:
https://github.com/unoplatform/Uno.Gallery/blob/afb97a5f511ef703d5676382e841f9295a3ba9ba/Uno.Gallery/Uno.Gallery.Mobile/iOS/Info.plist#L8

Not sure if this really matters though

**Review by kazo0 — APPROVED** [review]

**Comment by kazo0** [comment]
Very cool :) now to get the net6.0 apps included in the UI Tests!

File: samples/Uno.Toolkit.WinUI.Samples/Uno.Toolkit.WinUI.Samples.Mobile/Android/AndroidManifest.xml (hunk 1 above)
Comment (nickrandolph) [reply samples/Uno.Toolkit.WinUI.Samples/Uno.Toolkit.WinUI.Samples.Mobile/Android/AndroidManifest.xml]: Updated - thanks for noting this

File: samples/Uno.Toolkit.WinUI.Samples/Uno.Toolkit.WinUI.Samples.Mobile/iOS/Info.plist (hunk 2 above)
Comment (nickrandolph) [reply samples/Uno.Toolkit.WinUI.Samples/Uno.Toolkit.WinUI.Samples.Mobile/iOS/Info.plist]: Thanks

**Review by Xiaoy312 — APPROVED** [review]

**Comment by nickrandolph** [comment]
@kazo0 I noticed that some of the builds still have net6 disabled. We should sync so we can work through changes required for the build in order to integrate the net6 sample app

**Comment by kazo0** [comment]
@nickrandolph ah yes the Samples/Samples_WinUI jobs still have `DisableNet6MobileTargets: true`, I can remove that and then I guess we could just include the net6 sample heads in the custom Samples/Samples_WinUI solution configs. Or I guess we'd need it just for the Samples_WinUI config. Or we can have a standalone config for the net6 samples.

**Comment by nickrandolph** [comment]
@kazo0 did we ever manage to get the build sorted out for this?

**Comment by kazo0** [comment]
@nickrandolph yeah I still need to get this fixed, I should probably redo the way we are splitting the projects, I shouldn't really be using the ApplicationPlatform to do this

**Comment by nickrandolph** [comment]
@kazo0 in extensions for the testharness project we've moved to net6+winui for the main samples and then we have a "legacy" folder which contains samples for xamarin+uwp. Do you think that approach would work here too?

**Comment by agneszitte** [comment]
@nickrandolph and @kazo0 what is the status of this PR please?

**Comment by kazo0** [comment]
Oh boy, didn't even realize we had this one, I'm actually working on doing this for the new canary work, it's almost done and I'll have a new PR

**Comment by nickrandolph** [comment]
Closing this as it sounds like it's been deprecated by the work @kazo0 has done

**Pull request closed without merging**